hooks/useApproval.ts

```diff
@@ -0,0 +1,71 @@
+import { ethers } from "ethers"
+import { useEffect, useState } from "react"
+import { erc20ABI, useAccount, useContract, useNetwork, useSigner } from "wagmi"
+
+const APPROVE_CALLBACK_STATUS = {
+  PENDING: 1,
+  FAIL: 2,
+  FINISH: 3,
+}
+
+const useApprove = (contractAddress: string, sepnder: string) => {
+  const [status, setStatus] = useState<number>()
+  const { data } = useAccount()
+  const { data: signer } = useSigner()
+  const contract = useContract({
+    addressOrName: contractAddress,
+    contractInterface: erc20ABI,
+    signerOrProvider: signer,
+  })
+
+  useEffect(() => {
+    if (signer) {
+      const filter = contract.filters.Approval(data?.address, contractAddress, null)
+      contract.on(filter, (address: string, sepnder: string, value: ethers.BigNumber) => {
```

Comment:
I was looking up `.on` not knowing how to use it. I understand that it is to capture events when the chain or connected address changes. However, I could not figure out why `.on` is needed here.

https://github.com/tmm/wagmi/blob/9a5af437ba472775bab91cd6452b8a2fb6f8d506/packages/core/src/connectors/mock/provider.ts#L80

Reply:
hmm, actually not necessary cuz approval will be listened by useApproval.
but I introduced with two reason
1. APPROVE_CALLBACK_STATUS looks better when it has pending finish error rather than just pending error
2. isolate this hooks from useApproval. for that, it needed to have finish status. otherwise, it will be pending all the time.

wdyt ? should I remove or stay this listener

Reply:
My personal preference would be not to add it until it is needed.